### PR TITLE
Command to generate page template files

### PR DESCRIPTION
**Pull request by barbaraperic**
## Note

Command `npm run generate:page -- whatever-page-name ` generates:
- new folder `whatever-page-name` under `app/src `with the following files:` page.tsx`, `layout.tsx`, `utils/generateStructuredData.ts`
- it creates `whatever-page-name.md` under `content/pages`
- it also updates `paths.ts` with necessary changes when adding a page

## How to test
- run `npm run generate:page -- page-name` and check generated files
- run `http://localhost:3000/page-name` and check if the page has been created

### CODE REVIEW

File: templates/page.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { graphicsData } from '@/data/graphicsData'
+
+import { generateStructuredData } from './utils/generateStructuredData'
+
+const { header, seo } = attributes
+
+export const metadata = createMetadata({
+  seo,
+  path: PATHS.__PATH_NAME__.path,
+  useAbsoluteTitle: true,
```
Comment (mirhamasala) [inline templates/page.tsx:22]:
```suggestion
```

File: templates/data/data.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+// ! change naming
+export const data: DataProps[] = [
+  {
+    icon: CurrencyCircleDollar,
+    title: '',
+    description: '',
+  },
+]
```
Comment (mirhamasala) [inline templates/data/data.tsx:19]: Wouldn't include this file.

File: scripts/generatePageTemplates.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      console.error('Error reading URLs configuration file:', err)
+      return
+    }
+
+    const pageNameCamelCase = toCamelCase(pageName)
+    const newObject = `  ${pageNameCamelCase}: {
+      email: {
+        href: '#',
+      },
+    },`
```
Comment (mirhamasala) [inline scripts/generatePageTemplates.js:109]: Wouldn't include.

File: templates/page.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+})
+
+export default function __PAGE_NAME_PASCALCASE__() {
+  return (
+    <PageLayout>
+      <StructuredDataScript structuredData={generateStructuredData(seo)} />
+      <PageHeader
+        title={header.title}
+        description={header.description}
+        image={graphicsData.imageFallback}
```
Comment (mirhamasala) [inline templates/page.tsx:32]: Think of whether we only want to include properties that are required.

**Review by mirhamasala — COMMENTED** [review]
Super cool! ✨ As discussed, let's see if we can:

- Run the linter
- Add test pages
- Sort paths.ts
- `scripts/generatePageTemplateHelpers`

**Review by mirhamasala — APPROVED** [review]
@barbaraperic I think this is brilliant and is going to be so rad to use. ✨

- `helpers.js` still needs renaming
- Any luck on alphabetizing and formatting `paths.ts`? (Otherwise, lets add the newly generated paths to the end.)
- Can we add a section to the README?
- The test template will have to change a bit after we merge this [>>](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/308/commits/e999e512eb140769857571875247299f5e81cce2)

Thank you. 🙏🏼

**Comment by barbaraperic** [comment]
> * `helpers.js` still needs renaming
✅

> * Any luck on alphabetizing and formatting `paths.ts`? (Otherwise, lets add the newly generated paths to the end.)
✅

> * Can we add a section to the README?
✅

> * The test template will have to change a bit after we merge this [>>](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/308/commits/e999e512eb140769857571875247299f5e81cce2)
✅

Thank you 🙏🏼

**Pull request merged**